ui/js/signup.js

```diff
@@ -0,0 +1,358 @@
+                'usernameavailable': {
+                    'method': function(value, element) {
+                        var response = false;
+                        isUserNameAvailable(value, function(available) {
+                            response = available;
+                            // Show the available icon if the username is available, otherwise show the unavailable icon
+                            if (available) {
+                                $('#signup-createaccount-username-available').removeClass('fa-times').addClass('fa-check');
+                            } else {
+                                $('#signup-createaccount-username-available').removeClass('fa-check').addClass('fa-times');
+                            }
+                        });
+                        return response;
+                    },
+                    'text': oae.api.i18n.translate('__MSG__THIS_USERNAME_HAS_ALREADY_BEEN_TAKEN__')
+                }
+            },
+            'submitHandler': createUser
+        };
+
+        oae.api.util.validation().validate($('#signup-createaccount-form'), validateOpts);
+    };
+
+    /**
+     * Set up a reCaptcha container that will used to verify
+     * that the current user is a real human
+     */
+    var setUpReCaptcha = function() {
+        if (recaptchaEnabled) {
+            // Only require the recaptcha library when recaptcha has been enabled
+            require(['//www.google.com/recaptcha/api/js/recaptcha_ajax.js'], function() {
+                var captchaContainer = $('#signup-createaccount-captcha-container')[0];
+                Recaptcha.create(recaptchaPublicKey, captchaContainer, {theme: 'custom'});
+            });
+
+            // Hide the Recaptcha error when text is entered
+            $('#recaptcha_response_field').on('keyup', hideRecaptchaError);
+
+            // Add a consistent height to the bottom row containers
+            $('#signup-createaccount-usercheck-container').addClass('signup-createaccount-captcha-create-has-captcha');
+        } else {
+            $('#signup-createaccount-captcha-container').hide();
+        }
+    };
+
+    /**
+     * Initialize the signup options template
+     */
+    var renderSignUpOptions = function() {
+        // Render the signup options
+        oae.api.util.template().render($('#signup-options-template'), {
+            'authStrategyInfo': authStrategyInfo,
+            'invitationInfo': invitationInfo,
+            'recaptchaEnabled': recaptchaEnabled,
+            'redirectUrl': signUpRedirectUrl,
+            'termsAndConditionsEnabled': termsAndConditionsEnabled
+        }, $('#signup-options-container'));
+
+        // For IE9, everything floats so we can't get height. Give the main container as much height
+        // as it needs for the create account form
+        $('.ie-lt10 #signup-options-local:visible').each(function() {
+            $('#signup-options-container').css('height', $(this).height());
```

Comment:
I forgot about that trick to get height when all your children float.

Unfortunately, I don't think it will be enough because we seem to have an explicit height if we don't have flexbox to manage our height for us. That means we'd have to set `height: 100%` on the container *and* all its parents all the way up to `html`, however if the page contents overflows the 100% height of the viewport, it messes things up.